source/data-modeling/associations.txt

```diff
@@ -515,3 +515,410 @@ method to remove all embedded ``Album`` documents from the ``Band`` class:
+
+Association Behaviors
+---------------------
+
+When working with associations in {+odm+}, there are various behaviors that
+allow you to customize how associations are handled in your application. The following
+sections describe common association behaviors.
+
+Extensions
+~~~~~~~~~~
+
+Extensions allow you to add specific functionality to an association. You can
+define an extension on an association by specifying a block in the association
+definition, as shown in the following example:
+
+.. literalinclude:: /includes/data-modeling/association-behaviors.rb
+   :language: ruby
+   :start-after: # start-extensions
+   :end-before: # end-extensions
+   :emphasize-lines: 4-6
+
+Custom Association Names
+~~~~~~~~~~~~~~~~~~~~~~~~
+
+You can use the ``class_name`` macro to specify a custom class name for an
+association. This is useful when you want to use a different name for an
+association than the name of the class. The following example uses the
+``class_name`` macro to specify that an embedded association called ``records``
+represents the ``Album`` class:
+
+.. literalinclude:: /includes/data-modeling/association-behaviors.rb
+   :language: ruby
+   :start-after: # start-custom-name
+   :end-before: # end-custom-name
+
+Custom Keys
+~~~~~~~~~~~
+
+By default, {+odm+} uses the ``_id`` field of the parent class when looking up
+associations. You can specify a different field to use by specifying the
+``primary_key`` and ``foreign_key`` macros. The following example specifies a new
+primary and foreign key for the ``albums`` association on a ``Band`` class:
+
+.. literalinclude:: /includes/data-modeling/association-behaviors.rb
+   :language: ruby
+   :start-after: # start-custom-keys
+   :end-before: # end-custom-keys
+
+If you are specifying a ``has_and_belongs_to_many`` association, you can also
+set the ``inverse_primary_key`` and ``inverse_foreign_key`` macros. The
+``inverse_primary_key`` macro specifies the field on the local model that the
+remote model uses to look up the documents.
+The ``inverse_foreign_key`` macro specifies the field on the remote model
+that stores the values found in ``inverse_primary_key``.
+
+The following example specifies a new primary and foreign key for the
+``Band`` and ``Members`` classes in a ``has_and_belongs_to_many`` association:
+
+.. literalinclude:: /includes/data-modeling/association-behaviors.rb
+   :language: ruby
+   :start-after: # start-custom-inverse-keys
+   :end-before: # end-custom-inverse-keys
+   :emphasize-lines: 9, 20
+
+Custom Scopes
+~~~~~~~~~~~~~
+
+You can set a specific scope on an association by using the ``scope`` parameter.
+The ``scope`` parameter determines which documents {+odm+} considers to be part
+of an association. A scoped association returns only documents that match the
+scope conditions when queried. You can set the ``scope`` to either a ``Proc`` with an arity
+of zero or a ``Symbol`` that references a named scope on the associated model.
```

Comment:
q: i'm assuming this will make sense to someone who knows mongoid?

Reply:
Yeah, I had to look it up but it's apparently common db/ruby terms